src/chef-mover/src/mover_global_admins_user_addition_callback.erl

```diff
@@ -0,0 +1,106 @@
+%% -*- erlang-indent-level: 4;indent-tabs-mode: nil; fill-column: 92 -*-
+%% ex: ts=4 sw=4 et
+%% @author Steven Danna <steve@chef.io>
+%% @copyright 2015 Chef Software, Inc.
+%%
+%% The global_admins_user_addition addition callback adds the user
+%% group from a given organization into the ORGNAME_global_admins
+%% group for that organization.
+%%
+%% The ORGNAME_global_admins group has a signle purpose: providing
+%% READ access to other user objects in ORGNAME.  Thus, the only
+%% change that this migration produces is that users that share an org
+%% can now see each others user objects.
+%%
+%% This change is necessary to facilitate the use of chef-vault.
+%%
+-module(mover_global_admins_user_addition_callback).
+
+-export([
+         migration_init/0,
+         migration_complete/0,
+         migration_type/0,
+         supervisor/0,
+         migration_start_worker_args/2,
+         migration_action/2,
+         next_object/0,
+         error_halts_migration/0,
+         reconfigure_object/2,
+         needs_account_dets/0
+        ]).
+
+-include("mover.hrl").
+
+-record(org, {name, id, authz_id}).
+-record(group, {name, id, authz_id}).
+
+migration_init() ->
+    mv_oc_chef_authz_http:create_pool(),
+    mover_transient_migration_queue:initialize_queue(?MODULE, all_orgs()).
```

Comment:
Migrating involves:

- 2 reads from the opscode_chef database
- 1 write to the bifrost API.  The API request is a no-op if it has already been completely previously.

Thus, I think that re-running the full migration should be safe.  That said, it will be slower than necessary and cause extra load on the database. The only way I can think of to get the "unmigrated orgs" is almost as much work as the migration since we have to:

- look up the authz_id of the ORGNAME_global_admins group
- look up the authz_id of the "users" group for ORGNAME
- ask biforst if the users group is a member of the ORGNAME_global_admins group 

Since the only write operation in the migration is a no-op if the users group is already in the ORGNAME_global_admins group, I think that rerunning the migration is actually cheaper than figuring out which orgs haven't been migrated. Perhaps there is some shorter path that I'm missing.

In general, I think we need to fix up the migration state tracking functionality.  As far as I can tell, none of that works now that orgs are in sql.  It is also a bit suboptimal as it requires 1 row per org per migration.  Perhaps something closer to how sqitch tracks migrations but on a per-org basis would be better.